File: frontend/src/types/test.run.device.status.ts
import IDeviceData from "./device";

export default interface ITesRunDeviceStatusData {
    id?: number,
    testRunId: number,
    deviceId: number,
    device: IDeviceData,
    startupTime: number,
    histAvgStartupTime: number,
    createdAt: Date,
    updatedAt: Date,
    deletedAt: Date,
}

File: storage/models/test_run_device_status.go
package models

type TestRunDeviceStatus struct {
	Model
	TestRunID   uint    `json:"testRunId"`
	DeviceID    uint    `json:"deviceId"`
	Device      *Device `json:"device"`
	StartupTime uint    `json:"startupTime"`

	// calculated fields
	HistAvgStartupTime uint `db:"-" gorm:"-:all" json:"histAvgStartupTime"`
}
